src/main/java/org/jenkinsci/plugins/docker/workflow/WithContainerStep.java

```diff
@@ -124,7 +124,7 @@ private static void destroy(String container, Launcher launcher, Node node, EnvV
 
         @Override public boolean start() throws Exception {
             EnvVars envReduced = new EnvVars(env);
-            EnvVars envHost = computer.getEnvironment();
+            EnvVars envHost = computer.buildEnvironment(TaskListener.NULL);
```

Comment:
I think the issue may be related to `envHost` being used for multiple purposes. The usage from `envReduced` is rather tricky to test due to its nature, but the other usages should be straightforward to test.